src/Components/ServiceSelectionScene/ServiceSelectionTabsScene.tsx

```diff
@@ -0,0 +1,297 @@
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState, SceneQueryRunner } from '@grafana/scenes';
+import React, { useRef } from 'react';
+import { Icon, Popover, PopoverController, Tab, TabsBar, Tooltip, useStyles2 } from '@grafana/ui';
+import { GrafanaTheme2, LoadingState, SelectableValue } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { SERVICE_NAME, SERVICE_UI_LABEL } from '../../services/variables';
+import { truncateText } from '../../services/text';
+import { rest } from 'lodash';
+import { ServiceSelectionScene } from './ServiceSelectionScene';
+import { getSceneQueryRunner } from '../../services/panel';
+import { buildResourceQuery } from '../../services/query';
+import { TabPopoverScene } from './TabPopoverScene';
+import { getDataSourceVariable, getServiceSelectionPrimaryLabel } from '../../services/variableGetters';
+import { getFavoriteTabsFromStorage, removeTabFromLocalStorage } from '../../services/store';
+
+export interface TabOption extends SelectableValue<string> {
+  label: string;
+  value: string;
+  active?: boolean;
+  counter?: number;
+  saved?: boolean;
+  savedIndex?: number;
+}
+
+export interface ServiceSelectionTabsSceneState extends SceneObjectState {
+  tabOptions: TabOption[];
+  showPopover: boolean;
+  $labelsData: SceneQueryRunner;
+  popover?: TabPopoverScene;
+}
+
+interface LabelOptions {
+  label: string;
+  cardinality: number;
+}
+
+export class ServiceSelectionTabsScene extends SceneObjectBase<ServiceSelectionTabsSceneState> {
+  constructor(state: Partial<ServiceSelectionTabsSceneState>) {
+    super({
+      showPopover: false,
+      $labelsData: getSceneQueryRunner({
+        queries: [buildResourceQuery('', 'detected_labels')],
+        runQueriesMode: 'manual',
+      }),
```

Comment:
Maybe with a `/label/LABELNAME/values` API call this would not lead to the different cardinality count that we saw.

Reply:
Feel free to try that in a followup PR, not here.

Reply:
We'd need to query each label individually though?